torch/csrc/jit/script/concrete_module_type.cpp

```diff
@@ -27,6 +29,18 @@ ClassTypePtr ConcreteModuleTypeBuilder::createTypeFromThis() const {
     cls->addAttribute(name, type, isParameter);
   }
 
+  for (const auto& pr : constants_) {
+    const auto& name = pr.first;
+    const auto& val = pr.second.v_;
+    auto match = tryToInferType(val);
+    if (!match.success()) {
+      TORCH_CHECK(false, "We need to infer the type of constant to convert the python value to IValue, but failed to infer type of ", py::str(val), "\n:", match.reason());
```

Comment:
This should be a `TORCH_INTERNAL_ASSERT` instead of a `TORCH_CHECK`, because our system is responsible for validating the precondition (that python value is convertible to a constant value).